### PR TITLE
HBASE-23069 periodic dependency bump for Sep 2019

### PR DESCRIPTION
**Pull request by saintstack**
caffeine: 2.6.2 => 2.8.1
commons-codec: 1.10 => 1.13
commons-io: 2.5 => 2.6
disrupter: 3.3.6 => 3.4.2
httpclient: 4.5.3 => 4.5.11
httpcore: 4.4.6 => 4.4.13
jackson: 2.9.10 => 2.10.1
jackson.databind: 2.9.10.1 => 2.10.1
jetty: 9.3.27.v20190418 => 9.3.28.v20191105
jruby: 9.1.17.0 => 9.2.9.0
protobuf.plugin: 0.5.0 => 0.6.1
zookeeper: 3.4.10 => 3.4.14
slf4j: 1.7.25 => 1.7.30
rat: 0.12 => 0.13
asciidoctor: 1.5.5 => 1.5.8.1
asciidoctor.pdf: 1.5.0-alpha.15 => 1.5.0-rc.2
error-prone: 2.3.3 => 2.3.4

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   2m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m 23s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   3m 34s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   5m 12s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m 20s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 25s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 28s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 28s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  shadedjars  |   5m 23s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  19m 46s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   3m 22s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 230m  2s |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 53s |  The patch does not generate ASF License warnings.  |
|  |   | 299m 31s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.rest.client.TestRemoteTable |
|   | hadoop.hbase.client.TestShell |
|   | hadoop.hbase.client.TestTableShell |
|   | hadoop.hbase.client.TestReplicationShell |
|   | hadoop.hbase.client.TestShellNoCluster |
|   | hadoop.hbase.client.rsgroup.TestShellRSGroups |
|   | hadoop.hbase.client.TestQuotasShell |
|   | hadoop.hbase.client.TestAdminShell |
|   | hadoop.hbase.client.TestAdminShell2 |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1082/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1082 |
| Optional Tests | dupname asflicense javac javadoc unit shadedjars hadoopcheck xml compile |
| uname | Linux d0a2fb6aa58e 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1082/out/precommit/personality/provided.sh |
| git revision | branch-2 / ac852c2b66 |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1082/1/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1082/1/testReport/ |
| Max. process+thread count | 5619 (vs. ulimit of 10000) |
| modules | C: . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1082/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by saintstack** [comment]
New push walks back the ruby change and the httpclient changes.

The first fails shell tests because missing method:
```

[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 21.575 s <<< FAILURE! - in org.apache.hadoop.hbase.client.TestShell
[ERROR] org.apache.hadoop.hbase.client.TestShell.testRunShellTests  Time elapsed: 0.558 s  <<< ERROR!
org.jruby.embed.EvalFailedException: (LoadError) load error: hbase_constants -- java.lang.NoSuchMethodError: org.jcodings.Encoding.caseMap(Lorg/jcodings/IntHolder;[BLorg/jcodings/IntHolder;I[BII)I
	at org.apache.hadoop.hbase.client.TestShell.testRunShellTests(TestShell.java:40)
Caused by: org.jruby.exceptions.LoadError: (LoadError) load error: hbase_constants -- java.lang.NoSuchMethodError: org.jcodings.Encoding.caseMap(Lorg/jcodings/IntHolder;[BLorg/jcodings/IntHolder;I[BII)I
Caused by: java.lang.NoSuchMethodError: org.jcodings.Encoding.caseMap(Lorg/jcodings/IntHolder;[BLorg/jcodings/IntHolder;I[BII)I
	at org.apache.hadoop.hbase.client.TestShell.testRunShellTests(TestShell.java:40)
```

httpclient update will remove any '//' from our encoded REST URLs which is messing up our parse.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   3m  3s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 36s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   3m 34s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   5m 20s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m 19s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 19s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m 54s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m 54s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  shadedjars  |   5m 25s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  19m 38s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   3m 18s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 204m 19s |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 37s |  The patch does not generate ASF License warnings.  |
|  |   | 272m 53s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.security.provider.TestCustomSaslAuthenticationProvider |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1082/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1082 |
| Optional Tests | dupname asflicense javac javadoc unit shadedjars hadoopcheck xml compile |
| uname | Linux 0c03b78b67a8 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1082/out/precommit/personality/provided.sh |
| git revision | branch-2 / 62e340901f |
| Default Java | 1.8.0_181 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1082/2/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1082/2/testReport/ |
| Max. process+thread count | 4722 (vs. ulimit of 10000) |
| modules | C: . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1082/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by saintstack** [comment]
Looking for a +1 here please. The failed test is known broken being worked on currently. Thanks.

File: pom.xml [hunk 1]
```diff
@@ -1648,6 +1652,16 @@
         <artifactId>hbase-zookeeper</artifactId>
         <groupId>org.apache.hbase</groupId>
         <version>${project.version}</version>
+        <exclusions>
+          <exclusion>
+            <groupId>com.google.code.findbugs</groupId>
+            <artifactId>jsr305</artifactId>
```
Comment (ndimiduk) [inline pom.xml]: How you conclude these exclusions are necessary? We have a static check someplace?

**Comment by ndimiduk** [comment]
Shouldn't this be into `master` ?

File: pom.xml (hunk 1 above)
Comment (saintstack) [reply pom.xml]: We have an errorprone rule that warns of jsr305 inclusions when release profile is in place.

**Comment by saintstack** [comment]
Will forward port to Master.

On whether there are failures attributable to these lib updates, dunno. Maybe there are. Will be looking at flakey list next. Will keep this in mind.

Thanks for the review.

**Pull request merged**